src/librustdoc/html/render/context.rs

```diff
@@ -372,10 +372,17 @@ impl<'tcx> FormatRenderer<'tcx> for Context<'tcx> {
         } = options;
 
         let src_root = match krate.src {
-            FileName::Real(ref p) => match p.local_path().parent() {
-                Some(p) => p.to_path_buf(),
-                None => PathBuf::new(),
-            },
+            FileName::Real(ref p) => {
+                if let Some(local_path) = p.local_path() {
+                    match local_path.parent() {
+                        Some(p) => p.to_path_buf(),
+                        None => PathBuf::new(),
+                    }
+                } else {
+                    // Somehow we got the filename from the metadata of another crate, should never happen
+                    PathBuf::new()
```

Comment:
This comment is definitely wrong. Rustdoc links to non-local crates all the time, whenever an item is re-exported across crates.

Reply:
This comment is misleading. local_path() will still be `Some()` if that path from an external crate hasn't been remapped. Same for other places